docker/Dockerfile.base

```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:16.04
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Update apt-get
+RUN apt-get -y update && \
+  # Install apt-utils
+  apt-get -y --no-install-recommends install apt-utils && \
+  # Main Build Dependencies
+  apt-get -y --no-install-recommends install git build-essential cmake clang-3.8 \
+  g++-4.8 wget ca-certificates libpcre3 libpcre3-dev python unzip \
+  # Docs Build Dependencies
+  python-sphinx doxygen graphviz \
+  # Json Platform Build Dependencies
+  pkg-config libjson0 libjson0-dev \
+  # Imraa Platform Build Dependencies
+  libudev-dev \
+  # Doc Build Dependencies
+  python-sphinx doxygen graphviz
+
+# Set library download folder to /opt
+WORKDIR /opt
+
+# Swig Build Dependencies
+RUN wget -q http://iotdk.intel.com/misc/tr/swig-3.0.10.tar.gz && \
```

Comment:
while we are at it - it would be worth it to split out the version string into a separate var, so upon update one doesn't have to update in zillion places [and surely miss one of them] :)